commit bff2a944889aa8b636a06ae5d80d2e0921837168
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 16:10:10 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/lamarck/patch-aclocalm4-libcurl.diff

diff --git a/aclocal.m4 b/aclocal.m4
index cc92d20c3..a86745216 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -887,15 +887,10 @@ AC_DEFUN([MP_LIBCURL_FLAGS],[
 	fi
 
 	CFLAGS_LIBCURL=$($CURL_CONFIG --cflags)
-	if test "x$curlprefix" = "x"; then
-		# System curl-config emits absurd output for --libs
-		# See rdar://7244457
-		LDFLAGS_LIBCURL="-lcurl"
-	else
-		# Due to a bug in dist, --arch flags are improperly supplied by curl-config.
-		# Get rid of them.
-		LDFLAGS_LIBCURL=$($CURL_CONFIG --libs | [sed 's/-arch [A-Za-z0-9_]* //g'])
-	fi
+	LDFLAGS_LIBCURL=$($CURL_CONFIG --libs)
+	# prefer to use libcurl-gnutls.so which can't cause openssl version clashes
+	AC_CHECK_LIB([curl-gnutls], [curl_global_init],[
+		LDFLAGS_LIBCURL=-lcurl-gnutls])
 
 	AC_SUBST(CFLAGS_LIBCURL)
 	AC_SUBST(LDFLAGS_LIBCURL)
